platform/src/ToolsManager.js

```diff
@@ -31,10 +31,8 @@ class ToolManager {
                 
                 let toolUrl = new Object();
                 toolUrl.id = ""; // Populate when tool is fetched
-                if (this.isValidUrl(url)){
-                    toolUrl.url = url;
-                }
-                else{
+                if (this.isUrlPlaceHolder(url)){
```

Comment:
But this also returns true for `{{ID-...}}` rewrite requests, so those would be treated as if they are `{{BASE-URL}}` occurrences. That seems wrong.

Reply:
Yes, when `{{ID-***}}` placeholders are interpolated, then they won't get to this point at all, so that's not a problem.

**However**, when they aren't then they get here and there really isn't any valid substitution -- this is an expected situation not an error. Substituting as if it was a `{{BASE-URL}}` is definitely not valid. What should happen is that the case is recognised and ignored, with the containing activity being disabled. I think `EducationPlatform` manages that, in fact.

Reply:
We don't currently have a check at https://github.com/mdenet/educationplatform/blob/fe88277fd9cca3e4b9adbfcc4448e10423c828cd/platform/src/ToolsManager.js#L58

As you can see. If the `isValidURL()` method only returns true for actual URLs and not placeholders, then we can add a guard to skip `{{ID-...}}` at this stage. When they get interpolated, they will be a valid URL and can pass through.

Reply:
An error is pushed to the UI in `setToolsUrls()` when a candidate URL is neither a placeholder, nor a valid URL. Other cases it can throw an error is perhaps when doing string manipulation.

Reply:
Yes. The point I'm making is that unresolved `{{ID-***}}` tool URLs shouldn't lead to such an error being pushed, but should also not be rewritten as if they were `{{BASE-URL}}:port` URLs instead.

Reply:
As discussed `isUrlPlaceHolder() ` is refactored into `isBaseUrlPlaceHolder()` and `isIDPlaceHolder()` and logic around extracting URL tail is improved.